app/src/main/java/org/wikipedia/readinglist/ReadingListsFragment.kt

```diff
@@ -486,6 +580,108 @@ class ReadingListsFragment : Fragment(), SortReadingListsDialog.Callback, Readin
+        get() {
+            return displayedLists.let {
+                displayedLists.filterIsInstance<ReadingList>()
+                    .filter { it.selected }
+                    .onEach { it.selected = false }
+            }
+        }
+
+    private fun updateMenuItemsAlpha(alpha: Int, menu: Menu) {
+        val deleteItem = menu.findItem(R.id.menu_delete_selected)
+        val exportItem = menu.findItem(R.id.menu_export_selected)
+        val exportItemTitleColor = ResourceUtil.getThemedColor(requireContext(), R.attr.colorAccent)
+        val exportItemAlphaColor = ColorUtils.setAlphaComponent(exportItemTitleColor, alpha)
+        val spanString = SpannableString(exportItem.title.toString())
+        spanString.setSpan(ForegroundColorSpan(exportItemAlphaColor), 0, spanString.length, 0)
+        exportItem.title = spanString
+        deleteItem.icon?.alpha = alpha
+    }
+
+    private inner class MultiSelectCallback : MultiSelectActionModeCallback() {
+        override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+            super.onCreateActionMode(mode, menu)
+            mode.menuInflater.inflate(R.menu.menu_action_mode_reading_lists, menu)
+            actionMode = mode
+            val deleteItem = menu.findItem(R.id.menu_delete_selected)
+            val deleteIconColor = ResourceUtil.getThemedColorStateList(requireContext(), R.attr.colorError)
+            MenuItemCompat.setIconTintList(deleteItem, deleteIconColor)
+            return true
+        }
+
+        override fun onActionItemClicked(mode: ActionMode, menuItem: MenuItem): Boolean {
+            when (menuItem.itemId) {
+                R.id.menu_delete_selected -> {
+                    onDeleteSelected()
+                    finishActionMode()
+                    return true
+                }
+                R.id.menu_export_selected -> {
+                    if (selectedListsCount == 0) {
+                        Toast.makeText(context, getString(R.string.reading_lists_export_select_lists_message),
+                            Toast.LENGTH_SHORT).show()
+                        return true
+                    }
+                    ReadingListsExportImportHelper.exportLists(activity as BaseActivity, selectedLists)
+                    finishActionMode()
+                    return true
+                }
+                R.id.menu_check_all -> {
+                    selectAllLists()
+                    mode.menu.findItem(R.id.menu_check_all).isVisible = false
+                    mode.menu.findItem(R.id.menu_uncheck_all).isVisible = true
+                }
+                R.id.menu_uncheck_all -> {
+                    unselectAllLists()
+                    mode.menu.findItem(R.id.menu_uncheck_all).isVisible = false
+                    mode.menu.findItem(R.id.menu_check_all).isVisible = true
+                }
+            }
+            return false
+        }
+
+        override fun onDeleteSelected() {
```

Comment:
The currently selected items will be deselected and exit the action mode after the app shows the "Delete selected list(s)?" dialog. Would it be possible to keep the action mode and only exit the action mode after users manually close it?